examples/hyperbee.js

```diff
@@ -0,0 +1,160 @@
+'use strict'
+
+process.env.DEBUG = 'bfx:*'
+
+const path = require('path')
+
+const async = require('async')
+const hypercore = require('hypercore')
+const Hyperbee = require('hyperbee')
+
+const replicate = require('@hyperswarm/replicator')
+
+const { Candle } = require('bfx-api-node-models')
+const { SYMBOLS, TIME_FRAMES } = require('bfx-hf-util')
+const EMAStrategy = require('bfx-hf-strategy/examples/ema_cross')
+
+const { onCandle, onStart, onEnd } = require('../lib/ws_events')
+const initState = require('../lib/init_state')
+
+const hopts = {
+  overwrite: true
+}
+
+const hbOpts = {
+  valueEncoding: 'json'
+}
+
+const rawCandleData = require('./btc_candle_data.json')
+
+rawCandleData.sort((a, b) => {
+  return a[0] - b[0]
+})
+
+const kp = 'candles!5m!'
+const from = rawCandleData[0][0]
+const to = rawCandleData[999][0]
+
+console.log('from:', from, 'to:', to)
+
+let feed
+let localFeed
+let db
+let localDb
+async.auto({
+  setupCores: (next) => {
+    // server has 1000 entries
+    feed = hypercore(path.join(__dirname, 'dbs', 'BTCUSD'), hopts)
+    db = new Hyperbee(feed, hbOpts)
+
+    // lets share it
+    db.feed.ready(() => {
+      replicate(feed, { announce: true, live: true, lookup: false })
+
+      // localcopy will have a frame of it
+      // lets sync some data from the middle
+      localFeed = hypercore(path.join(__dirname, 'dbs', 'BTCUSD_frame'), feed.key, {
+        ...hopts,
+        sparse: true
+      })
+
+      localDb = new Hyperbee(localFeed, hbOpts)
+
+      localDb.feed.ready(next)
+    })
+  },
+
+  setupRep: ['setupCores', (res, next) => {
+    // remove me in new hypercore version
+
+    localFeed.once('peer-add', () => {
+      next()
+    })
+
+    replicate(localFeed, { lookup: true, live: false })
+  }],
+
+  prepareCandles: ['setupCores', async (res) => {
+    // feed data into the data source / server
+
+    const batch = db.batch()
+    for (let i = 0; i < rawCandleData.length; i++) {
```

Comment:
Maybe sort from oldest to most recent before pushing to hyperbee? I usually like to do that whenever manipulating candles

```
rawCandleData = rawCandleData.sort((x, y) => x.mts - y.mts)
```

Reply:
its done in L30. 

And here is the cool thing: because of the btree we don't need to sort any more. Hyperbee is sorting the data for us. That means when we incrementally sync candles from lets say May to June, and afterwards sync February, the data is still accessible in the right order (was not the case with plain hypercores)

That's why we use an extension here

Reply:
Ok gotcha! that makes total sense